docs/data/data-grid/column-groups/CollapsibleColumnGroups.js

```diff
@@ -0,0 +1,123 @@
+import * as React from 'react';
+import { styled } from '@mui/material/styles';
+import Box from '@mui/material/Box';
+import IconButton from '@mui/material/IconButton';
+import {
+  DataGrid,
+  gridColumnVisibilityModelSelector,
+  useGridApiContext,
+} from '@mui/x-data-grid';
+import ArrowRightIcon from '@mui/icons-material/ArrowRight';
+import ArrowLeftIcon from '@mui/icons-material/ArrowLeft';
+
+const COLLAPSIBLE_COLUMN_GROUPS = {
+  character: ['lastName', 'age'],
+  naming: ['lastName'],
+};
+
+const ColumnGroupRoot = styled('div')({
+  overflow: 'hidden',
+  display: 'flex',
+  alignItems: 'center',
+});
+
+const ColumnGroupTitle = styled('span')({
+  overflow: 'hidden',
+  textOverflow: 'ellipsis',
+});
+
+function CollapsibleHeaderGroup({ groupId, headerName }) {
+  const apiRef = useGridApiContext();
+  const columnVisibilityModel = gridColumnVisibilityModelSelector(apiRef);
+
+  if (!groupId) {
+    return null;
+  }
+
+  const isCollapsible = Boolean(COLLAPSIBLE_COLUMN_GROUPS[groupId]);
+  const isGroupCollapsed = COLLAPSIBLE_COLUMN_GROUPS[groupId].every(
+    (field) => columnVisibilityModel[field] === false,
+  );
+
+  return (
+    <ColumnGroupRoot>
+      <ColumnGroupTitle>{headerName ?? groupId}</ColumnGroupTitle>{' '}
+      {isCollapsible && (
+        <IconButton
+          sx={{ ml: 0.5 }}
+          onClick={() => {
+            const newModel = { ...columnVisibilityModel };
+            COLLAPSIBLE_COLUMN_GROUPS[groupId].forEach((field) => {
+              newModel[field] = !!isGroupCollapsed;
+            });
+            apiRef.current.setColumnVisibilityModel(newModel);
+          }}
+        >
+          {isGroupCollapsed ? (
```

Comment:
I would suggest to align this with the [Row grouping](https://deploy-preview-14818--material-ui-x.netlify.app/x/react-data-grid/row-grouping/)
expanded: arrow down
collapsed: arrow right
(like the first example in https://github.com/mui/mui-x/issues/14033)
This follows the second example, but I think that then we need other icons (like in that example) instead of simple arrows